diff -Naur trunk/dfu-programmer/src/arguments.c trunk.new//dfu-programmer/src/arguments.c
--- trunk/dfu-programmer/src/arguments.c        2010-12-20 17:24:22.503381444 +0100
+++ trunk.new//dfu-programmer/src/arguments.c   2010-12-20 17:18:36.174616768 +0100
@@ -75,8 +75,9 @@
     { "at90usb82",      tar_at90usb82,      adc_AVR,   0x2FF7, 0x03eb, 0x02000, 0x1000, true,  128, true,  false, 128, 0x0200 },
     { "atmega32u6",     tar_atmega32u6,     adc_AVR,   0x2FF2, 0x03eb, 0x08000, 0x1000, true,  128, true,  false, 128, 0x0400 },
     { "atmega32u4",     tar_atmega32u4,     adc_AVR,   0x2FF4, 0x03eb, 0x08000, 0x1000, true,  128, true,  false, 128, 0x0400 },
-    { "atmega32u2",     tar_atmega32u2,     adc_AVR,   0x2FF0, 0x03eb, 0x08000, 0x1000, true,  128, true, false,  128, 0x0400 },
+    { "atmega32u2",     tar_atmega32u2,     adc_AVR,   0x2FF0, 0x03eb, 0x08000, 0x1000, true,  128, true,  false, 128, 0x0400 },
     { "atmega16u4",     tar_atmega16u4,     adc_AVR,   0x2FF3, 0x03eb, 0x04000, 0x1000, true,  128, true,  false, 128, 0x0200 },
+    { "atmega8u2",      tar_atmega8u2,      adc_AVR,   0x2FEE, 0x03eb, 0x02000, 0x1000, true,  128, true,  false, 128, 0x0200 },
     { "at32uc3a0128",   tar_at32uc3a0128,   adc_AVR32, 0x2FF8, 0x03eb, 0x20000, 0x2000, false, 512, false, true,  0,   0      },
     { "at32uc3a1128",   tar_at32uc3a1128,   adc_AVR32, 0x2FF8, 0x03eb, 0x20000, 0x2000, false, 512, false, true,  0,   0      },
     { "at32uc3a0256",   tar_at32uc3a0256,   adc_AVR32, 0x2FF8, 0x03eb, 0x40000, 0x2000, false, 512, false, true,  0,   0      },
diff -Naur trunk/dfu-programmer/src/arguments.h trunk.new//dfu-programmer/src/arguments.h
--- trunk/dfu-programmer/src/arguments.h        2010-12-20 17:24:22.743408138 +0100
+++ trunk.new//dfu-programmer/src/arguments.h   2010-12-20 17:10:18.287982626 +0100
@@ -55,6 +55,7 @@
                     tar_atmega32u4,
                     tar_atmega32u2,
                     tar_atmega16u4,
+                    tar_atmega8u2,
                     tar_at32uc3b064,
                     tar_at32uc3b164,
                     tar_at32uc3b0128,
